sklearn/_build_utils/__init__.py

```diff
@@ -38,43 +57,50 @@ def build_from_c_and_cpp_files(extensions):
         extension.sources = sources
 
 
-def maybe_cythonize_extensions(top_path, config):
+def generate_cythonize_extensions(top_path, config):
```

Comment:
just ``cythonize_extensions`` seems a better name because it's what this function does. In numpy/scipy it's different because they also auto-generate some cython files.